### DISCUSSION TITLE
Variable 'object' does not exist

### DISCUSSION THREAD

**Discussion by ogurec-ogurec** [danirus/django-comments-xtd#402] [Q&A]
Hello, you made a great app, thanks for your work.

I had a couple of questions: an error,

I tried django-comments-ink  to put it on my project, it seems that I connected everything, but when I go to the page with comments, there is an error:

```
TemplateSyntaxError at /posts/test-posts
Variable 'object' does not exist
```

```
Request Method: | GET
http://127.0.0.1:8000/posts/test-posts
4.2.21
TemplateSyntaxError
Variable 'object' does not exist
C:\Users\John\AppData\Local\Programs\Python\Python311\Lib\site-packages\django_comments_ink\templatetags\comments_ink.py, line 1135, in resolve_ctype_and_object_pk
C:\Users\John\AppData\Local\Programs\Python\Python311\python.exe
3.11.5

Error during template rendering
In template my-blog\templates\base.html, error at line 0

Variable 'object' does not exist
```
what could it be?

**Comment by ogurec-ogurec** [comment]
@danirus Danirus, pls help

**Comment by danirus** [comment]
It looks like there is a reference to an 'object' in the method `resolve_ctype_and_object_pk` in `comments_ink.py` that does not exist. The code in the repository does not show that line 1135 in `comments_ink.py` belongs to the method `resolve_ctype_and_object_pk`. I guess your code is old.

I played a lot in django-comments-ink with caching. It is reflected in the code in the `main` branch. When you vote for a comment, or react to a comment, or send a new one (by the way, sending comments work without problems in `main`), I tried to put it in the cache. I also added a module called `partial.py` that is retrieving the QuerySet from cache, and I know it does so even when there are new comments. It happens because the cache key that holds the QuerySet is not updated correctly when a new comment arrives. If you put the logging of django_comments_ink to "DEBUG" in the settings, you will see the messages that indicate that results are being retrieve from the cache. If you clear the cache (`dci_cache.clear()`) then you get the correct results. I am sorry it is unstable and unfinished.

**Reply by ogurec-ogurec** [reply]
thank,
